snapcraft.yaml

```diff
@@ -101,3 +101,84 @@ parts:
       snapcraftctl prime
       # ensure build-in tests are run
       cd ${SNAPCRAFT_PART_SRC} && make test TESTDIR=${SNAPCRAFT_PRIME}
+
+  opensc:
+    plugin: autotools
+    source: https://git.launchpad.net/ubuntu/+source/opensc
+    source-type: git
+    source-depth: 1
+    source-branch: applied/ubuntu/focal
+    autotools-configure-parameters:
+      - --host ${SNAPCRAFT_ARCH_TRIPLET}
+      - --prefix=/usr
+      - --sysconfdir=/etc/opensc
+      - --with-pkcs11-provider=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/pkcs11/p11-kit-client.so
+      - --libdir=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}
+      - --disable-man
+      - --disable-sm
+    build-environment:
+      - LDFLAGS: "-w -s"
+    override-build: |
+      snapcraftctl build
+      make install-strip DESTDIR=${SNAPCRAFT_PART_INSTALL}
+    stage:
+      - usr/bin/pkcs11-tool
+      - usr/lib/*/libopensc.so*
+
+  opensc-libp11:
+    plugin: autotools
+    source: https://git.launchpad.net/ubuntu/+source/libp11
+    source-type: git
+    source-depth: 1
+    source-branch: applied/ubuntu/focal
+    autotools-configure-parameters:
+      - --host ${SNAPCRAFT_ARCH_TRIPLET}
+      - --prefix=/usr
+      - --sysconfdir=/etc/opensc
+      - --with-pkcs11-module=/usr/lib/${SNAPCRAFT_ARCH_TRIPLET}/pkcs11/p11-kit-client.so
```

Comment:
Same here. `p11-kit-proxy.so` is actually the default on many distributions.

Reply:
I'm not sure there is good use case though for using `p11-kit-proxy.so` here.
- from within snap, we do not consider use case using any "actual" modules directly, it should always go through `p11-kit-client.so` to keep client<>service isolation.
- `p11-kit-client.so` does not handle cases to access two p11 server instances at the same time, as it takes config from env. So there is no actual way to connect to multiple p11-server instances using  `p11-kit-proxy.so` and configuring it to load `p11-kit-client.so` "multiple times".
- if we use  `p11-kit-proxy.so`, we will face added complexity, how to configure `/etc/pkcs11/modules/` for each snap. Should it be writable for each snap with pkcs11 plug? But snap should not be able to configure this globally, as we can have multiple pkcs11 slots, but given snap can have only some connected. So we would need to overlay it for each snap, while in 90% or more cases this will be just overkill for no value.
- using default `p11-kit-client.so` make it easy to consume for 90% cases, while snap can still override the default behaviour by using `--module` and load proxy instead.

I see the point for proxy on actual distributions, where you want to configure globally what modules to "chain" load by the proxy. But there you configure directly `opensc` and other modules. This won't be possible on core anyway. So it will not have desired benefit.